tests/test_all_rules.py

```diff
@@ -93,7 +93,7 @@ class TestThreatMappings(BaseRuleTest):
 
     def test_technique_deprecations(self):
         """Check for use of any ATT&CK techniques that have been deprecated."""
-        replacement_map = attack.techniques_redirect_map
+        replacement_map = attack.load_techniques_redirect()
```

Comment:
We should be able to load the `enterprise-attack.json` file and from the latest refreshed attack data and check if the `x_mitre_deprecated: true` value is set for a specific technique instead of saving and referencing web redirections for deprecated techniques.

Reply:
This is the current method in order to share the new technique in order to make replacing it simple. If we parse out the relationship from the attack json, then you are right, we can replace it.

Related to: https://github.com/elastic/detection-rules/pull/2343/files#r993896948

Reply:
I attempted to load the data locally and search for a relational key that would link revoked or deprecated techniques to the new technique IDs and came up short in doing so. From what I could find, the only way to establish this would be by name but this may not be reliable if names were to change.

<img width="1014" alt="Screen Shot 2022-10-18 at 10 33 15 AM" src="https://user-images.githubusercontent.com/99630311/196460890-2fe4c2b3-759d-4db2-81ef-d75048da356e.png">

Logical steps if technique ID changes:
1. Refresh and load new attack data
2. For all IDs, check if they exist with `x-mitre-deprecated` and `revoked` being None or False
3. If `x-mitre-deprecated` or `revoked` exists, use `name` as key and search for other entry
4. In other entry, `external_references.external_id` is the new technique ID

Logical steps if name changes:
1. Refresh and load new attack data
2. For all IDs, check if they exist with `x-mitre-deprecated` and `revoked` being None or False
3. If a name still exists but is not `revoked` or `x-mitre-deprecated` but does not match the rule use technique ID as key
4. Search for entry based on technique ID, pull new name from `name` key in entry

Unfortunately outside of this, there is no consistent ID to use for tracking the entire history of an ATT&CK object from this data source that I could find.

Reply:
- What are the odds that the technique ID and name both change? 
- Is it possible to use the old attack data before deleting (during refresh) in the event the ID or name changes to do a lookup?